src/io.c

```diff
@@ -16,4 +16,15 @@ u8 in8(u16 port)
                : "Nd"(port));
 
   return value;
+}
+
+void memset(u32 *dest, int val, u32 size)
```

Comment:
This is a great addition but I've already added it to string.c since I want to stick to C standard lib standards so this is now redundant. `io.c` is also only really made for communication with other parts of the system, not memory manipulation functions.